### ISSUE TITLE
MIDIOut doesn't work well in Windows (a problem occurring in MIDIClockOut)

### ISSUE THREAD

**Issue by ofeliayorquesta** [supercollider/supercollider#2599]
MIDIClockOut class does not work well if I try to send MidiClock messages through LoopBe, and to an external Daw, in order to slave the DAW into Supercollider's Tempo Clock. The next code crashes the interpreter:

m = MIDIClockOut("LoopBe Internal MIDI", "LoopBe Internal MIDI");
m.play

**Comment by telephon** [comment]
This might be an issue of the respective quark: https://github.com/crucialfelix/crucial-library
Feel free to reopen if it turns out to be an sclang issue ...

**Comment by telephon** [comment]
… which it probably is, but then we have a different diagnosis.

**Comment by jamshark70** [comment]
Yes, it probably is either an sclang bug or a Windows problem. I just read the code for `MIDIClockOut` -- it's short, and the only thing in it that would crash is when it calls the MIDIOut object.

My bad experience in the past was that MIDIOut to an inter-app interface in Windows was fine until the event density was too high. I suspect PortMIDI (or our use of it?) may not be properly threaded: if we try to send a second message before sclang has finished sending the previous one, it could go boom.

I don't know if Luis's changes improved that, though.
